File: test_api.js
const https = require('https');
const FormData = require('form-data');
const fetch = require('node-fetch');

const API_URL = 'https://health-detective-api.onrender.com';

async function testRoot() {
  console.log('Testing root endpoint...');
  try {
    const response = await fetch(`${API_URL}/`);
    const data = await response.json();
    console.log('Root endpoint response:', data);
  } catch (error) {
    console.error('Error testing root endpoint:', error.message);
  }
}

async function testDiseases() {
  console.log('Testing diseases endpoint...');
  try {
    const response = await fetch(`${API_URL}/diseases`);
    const data = await response.json();
    console.log('Diseases endpoint response (first item):', 
      Object.keys(data).length > 0 ? Object.entries(data)[0] : 'No data');
  } catch (error) {
    console.error('Error testing diseases endpoint:', error.message);
  }
}

async function testPredict() {
  console.log('Testing predict endpoint...');
  try {
    const formData = new FormData();
    formData.append('symptoms', 'fever');
    formData.append('symptoms', 'cough');
    
    const response = await fetch(`${API_URL}/predict`, {
      method: 'POST',
      body: formData
    });
    
    if (!response.ok) {
      throw new Error(`Server responded with status: ${response.status}`);
    }
    
    const data = await response.json();
    console.log('Predict endpoint response:', data);
  } catch (error) {
    console.error('Error testing predict endpoint:', error.message);
  }
}

async function runTests() {
  console.log('Starting API tests...');
  await testRoot();
  await testDiseases();
  await testPredict();
  console.log('Tests completed.');
}

runTests(); 

File: test_api.py
import requests
import json

API_URL = 'https://health-detective-api.onrender.com'

def test_root():
    print('Testing root endpoint...')
    try:
        response = requests.get(f'{API_URL}/')
        data = response.json()
        print('Root endpoint response:', data)
    except Exception as e:
        print(f'Error testing root endpoint: {str(e)}')

def test_diseases():
    print('Testing diseases endpoint...')
    try:
        response = requests.get(f'{API_URL}/diseases')
        data = response.json()
        if data:
            first_item = list(data.items())[0]
            print('Diseases endpoint response (first item):', first_item)
        else:
            print('No data returned from diseases endpoint')
    except Exception as e:
        print(f'Error testing diseases endpoint: {str(e)}')

def test_predict():
    print('Testing predict endpoint...')
    try:
        data = {'symptoms': ['fever', 'cough']}
        response = requests.post(f'{API_URL}/predict', data=data)
        if response.ok:
            result = response.json()
            print('Predict endpoint response:', result)
        else:
            print(f'Server responded with status: {response.status_code}')
            print('Response text:', response.text)
    except Exception as e:
        print(f'Error testing predict endpoint: {str(e)}')

def run_tests():
    print('Starting API tests...')
    test_root()
    test_diseases()
    test_predict()
    print('Tests completed.')

if __name__ == '__main__':
    run_tests() 